File: abc189/abc189_c/main.py
#!/usr/bin/env python3

import sys
sys.setrecursionlimit(10**7)

def main():
    N = int(input())
    A = tuple(map(int, input().split()))
    
    ans = 0
    for i in range(N):
        min_val = A[i]
        for j in range(i, N):
            min_val = min(min_val, A[j])
            score = min_val * (j-i+1)
            ans = max(ans, score)
    print(ans)
            


if __name__ == '__main__':
    main()

File: abc189/abc189_c/main.cpp
#include<iostream>
#include<vector>
using namespace std;

int main(){
    int N;
    vector<int> A;
    
    cin >> N;
    A.resize(N);
    for (int i = 0; i < N; ++i) cin >> A[i];

    int ans = 0;
    for (int i = 0; i < N; ++i) {
        int min_val = A[i];
        for (int j = i; j < N; ++j) {
            min_val = min(min_val, A[j]);
            int score = (j-i+1) * min_val;
            ans = max(ans, score);
        }
    }
    cout << ans << endl;
    return 0;
}